public/app/core/services/echo/Echo.ts

```diff
@@ -0,0 +1,99 @@
+import { EchoConsumer, EchoMeta, EchoEvent, EchoSrv } from './types';
+import { KeyValue, CircularVector } from '@grafana/data';
+
+interface EchoConfig {
+  // How often should metrics be reported
+  flushInterval: number;
+  // Enables debug mode
+  debug: boolean;
+
+  // Size of given metric type buffer
+  buffersSize: number;
+}
+
+/**
+ * Echo is a service for collecting metrics from Grafana client-app
+ * It collects metrics, distributes them across registered consumers and flushes once per configured interval
+ * It's up to the registered consumer to decide what to do with a given type of metric
+ */
+export class Echo implements EchoSrv {
```

Comment:
When I was reading this class I was thinking this is an Observable that `consumers` can subscribe too but maybe I'am a crazy RxJs fan ;)

Reply:
No, you are not :) I just wanted to keep it simple and not introduce anything that would blur the picture. But yes, Rx is an option here too